src/bitmap/backend/atomic_bitmap_mmap.rs

```diff
@@ -0,0 +1,608 @@
+// SPDX-License-Identifier: Apache-2.0 or BSD-3-Clause
+
+use crate::bitmap::{Bitmap, BitmapSlice, WithBitmapSlice};
+use crate::mmap::{BitmapMmap, BitmapReplace, NewBitmap};
+use crate::{Address, GuestAddress, GuestUsize};
+use std::ops::Index;
+use std::os::fd::{AsRawFd, BorrowedFd};
+use std::sync::atomic::{AtomicU8, Ordering};
+use std::sync::{Arc, RwLock};
+use std::{io, ptr};
+
+// Size in bytes of the `VHOST_LOG_PAGE`
+const LOG_PAGE_SIZE: usize = 0x1000;
```

Comment:
I wonder: If this is vhost related... Should it maybe instead live under the vhost repo? Are there other bits that may benefit from this being here instead of vhost?

Reply:
I don't know any other project that could use this, so I agree with you that it just can be part of vhost. I can include this as part of https://github.com/rust-vmm/vhost/pull/206 wdty?

@XanClic I'll respond here, but I'll address your comments in that PR, is that ok?

Reply:
> I don't know any other project that could use this, so I agree with you that it just can be part of vhost. I can include this as part of [rust-vmm/vhost#206](https://github.com/rust-vmm/vhost/pull/206) wdty?

sounds good to me!

Reply:
> I wonder: If this is vhost related... Should it maybe instead live under the vhost repo? Are there other bits that may benefit from this being here instead of vhost?

This is a good point. On first impression this seems to me to be the right place, but it's also perfectly fine in vhost if we don't think it would be useful to any other crate.

@alexandruag @bonzini @jiangliu @tkreuzer @roypat WDYT?

Reply:
From the description and @germag's comment above, it does sounds rather vhost specific to me (in my mind I for some reason thought it was actually moved to the vhost crate already based on the comments above, lol), so I would be inclined to include it in the vhost crate instead of here.